utils/filesystem/files.go

```diff
@@ -1223,8 +1223,10 @@ func (fs *VFS) unzip(ctx context.Context, source string, destination string, lim
 	if err != nil {
 		return
 	}
-	// List of file paths to return
 
+	var fileCounter int
```

Comment:
I would suggest to use an atomic counter like `line 1230` so that we have some protection if we want later to parallelise any of the code